app/notify_api_flask_app.py

```diff
@@ -0,0 +1,15 @@
+from flask import Flask
+
+
+class NotifyApiFlaskApp(Flask):
+    @property
+    def is_prod(self):
+        return self.config["NOTIFY_ENVIRONMENT"] in {"live", "production"}
+
+    @property
+    def is_test(self):
+        return self.config["NOTIFY_ENVIRONMENT"] == "test"
+
+    @property
+    def should_send_zendesk_alerts(self):
```

Comment:
yeah i did get that, but then my vscode setup is a bit hit and miss on lots of inference too. we'll just have to use github's ai thing to sort it out for us

Reply:
id much rather have this than something like the following tho

```py
from app.utils import should_send_zendesk_alerts
if should_send_zendesk_alerts(current_app):
  ...
```

Reply:
an alternative would be something like the following:

but i dont really like it because we'd still need to use current_app for some things, and this new class for other things which feels messy

```py
# app/__init__.py (or wherever)
class NotifyApi:
  @property
  def is_prod(self):
    return current_app.config['NOTIFY_ENVIRONMENT'] == 'production'

notify_api = NotifyApi()
```

```py
# regular code
from app import notify_api
if notify_api.is_prod: 
  ...
```

Reply:
We don't need to change it, don't worry. Those other options are just complicating things more than necessary I think.